website/content/docs/rfcs/signed-commits.mdx

```diff
@@ -0,0 +1,100 @@
+---
+sidebar_label: Signed commits
+description: |-
+  An OpenBao RFC for allowing users to contribute without GPG/SSH signing their commits.
+---
+
+# Discontinue enforcing signed commits
+
+**Status**: this was agreed upon in [Issue #399](https://github.com/openbao/openbao/issues/399)
+
+## Summary
+This RFC proposes discontinuing the enforcement of signed commits via branch
+protection rules in our GitHub repositories. The current practice requires every
+commit to be signed, with the intention of ensuring authenticity and authorship
+verification. However, this enforcement introduces complexities and barriers
+that may outweigh its perceived security benefits. The primary goal of this
+proposal is to streamline the contribution process, making it more accessible to
+casual contributors while maintaining adequate security practices through other
+means. This document outlines the rationale, potential impacts, and alternative
+solutions to ensure the integrity and security of our codebase.
+
+## Problem Statement
+The requirement for signed commits is intended to enhance the security of our
+codebase by verifying the identity of contributors. However, this policy has
+introduced several challenges:
+
+* **Onboarding Difficulty**: New contributors often face a steep learning curve
+  to configure commit signing correctly.
+* **Increased Overhead**: Regular contributors spend additional time ensuring
+  that their commits are signed, which can slow down the development process.
+* **Exclusion of Contributors**: Contributors without the technical know-how or
+  access to necessary tools are discouraged from contributing, which reduces
+  community engagement and diversity.
+* **Temporal Validity of Verification**: The validity of commit signatures is
+  temporal. If a contributor deletes their SSH key from GitHub's UI, previously
+  "verified" commits may become "unverified," compromising the long-term
+  reliability of signature verification.
+* **Rebase Merge Issues**: Ideally, we would like to rebase pull requests when
+  merging to maintain the original commit history of the PR. However, due to a
+  long-standing bug in GitHub, all signatures in individual commits become
+  invalidated during a rebase, causing them to appear as unverified.
```

Comment:
One minor additional nit is that we use squash merges, which re-signs the squashed version of the original commits in the series with GitHub's key, effectively deleting the contributor's own attestation.

Reply:
It is mentioned a bit below in the `Rationale` section that we use Squash Merge

Reply:
Oh, d'oh. ;-) Thanks! :D